### ISSUE TITLE
[FEA] Make NVTabular TensorFlow dataloader support TF Sequence Feature Columns

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA/NVTabular#664]
This is a feature request to make the NVTabular Tensorflow dataloader compatible with the TF Sequence feature columns.

## Context

We have currrently a custom [DenseFeatures](https://github.com/NVIDIA/NVTabular/blob/main/nvtabular/framework_utils/tensorflow/layers/embedding.py) layer implementation in the NVTabular TF dataloader, which does efficient retrieval of embedding values and supports multi-hot categorical values. But the DenseFeatures layer does not support TF Sequential feature columns, which are compatible only with the Keras [SequenceFeatures](https://www.tensorflow.org/api_docs/python/tf/keras/experimental/SequenceFeatures) layer.

- [sequence_categorical_column_with_identity](https://www.tensorflow.org/api_docs/python/tf/feature_column/sequence_categorical_column_with_identity)
- [sequence_categorical_column_with_hash_bucket](https://www.tensorflow.org/api_docs/python/tf/feature_column/sequence_categorical_column_with_hash_bucket)
- [sequence_categorical_column_with_vocabulary_file](https://www.tensorflow.org/api_docs/python/tf/feature_column/sequence_categorical_column_with_vocabulary_file)
- [sequence_categorical_column_with_vocabulary_list](https://www.tensorflow.org/api_docs/python/tf/feature_column/sequence_categorical_column_with_vocabulary_list)
- [sequence_numeric_column](https://www.tensorflow.org/api_docs/python/tf/feature_column/sequence_numeric_column)

## Requirements
**RQ1**. So, we should implement the analogous **SequenceFeatures** layer in the NVTabular Data loader, to provide to our users a drop-in replacement in their pipelines using Sequence feature columns and SequenceFeatures layer. As each sequence might have different length, the Data Loader should provide an option to return either return a [RaggedTensors](https://www.tensorflow.org/guide/ragged_tensor) or a dense Tensor with padding zeros with a pre-defined length.

**RQ2**. We should be able to return sequential feature columns when using the [make_feature_column_workflow()](https://github.com/NVIDIA/NVTabular/blob/main/nvtabular/framework_utils/tensorflow/feature_column_utils.py#L35) method for columns with multi-hot categorical values or sequences of categorical or numerical features in general.

## Example
I have made it available an example with TF Sequence feature columns and SequenceFeatures layer [here](https://gitlab-master.nvidia.com/gmoreira/gmoreira_pocs/-/tree/main/tf_sequence_featurecolumns_example).
